src/processor/logic.py

```diff
@@ -35,21 +38,108 @@ def check_existing_entries(model_column, column_values):
     )
 
 
+def get_wiki_page_title(url):
+    """ Get the title of a Wikimedia page, based on its URL.
+
+    Args:
+        url (str): url of Wikimedia page.
+
+    Returns:
+        str: title of the Wikimedia page
+    """
+    url_obj = urlsplit(url)
+    return unquote(
+        os.path.basename(url_obj.path)
+    )
+
+
+def get_wikimedia_call_info(url):
+    """ Determine the URL and parameters to use query the Wikimedia API about
+    info on a specific page. Used for non-Wikipedia pages where the wikipedia
+    client cannot be used to fetch info.
+
+    Args:
+        url (str): url of Wikimedia page.
+
+    Returns:
+        tuple: url and dict of parameters to use for the API call.
+    """
+    url_obj = urlsplit(url)
+    title = get_wiki_page_title(url)
+
+    call_params = {
+        'action': 'query',
+        'prop': 'extlinks',
+        'redirects': 'true',
+        'format': 'json',
+        'titles': title,
+    }
+    call_url = f'https://{url_obj.netloc}/w/api.php'
+
+    return call_url, call_params
+
+
+def get_wikipedia_references(url):
+    """ Get a list of references from a Wikipedia page.
+
+    Args:
+        url (str): url of Wikipedia page.
+
+    Returns:
+        list: references from the wikimedia page.
+    """
+    page_language = get_language_from_wiki_url(url)
+    page_title = get_wiki_page_title(url)
+
+    wikipedia.set_lang(page_language)
+
+    try:
+        page_obj = wikipedia.WikipediaPage(title=page_title)
+    except PageError:
+        page_obj = wikipedia.page(title=page_title)
+
+    return page_obj.references
+
+
+def get_non_wikipedia_references(url):
+    """ Get list of references from a Wikimedia page that is not Wikipedia.
+    e.g. wikisource.org, meta.wikimedia.org.
+
+    Args:
+        url (str): url of Wikimedia page.
+
+    Returns:
+        list: references/external links from the wikimedia page.
+    """
+    call_url, call_params = get_wikimedia_call_info(url)
+    response = requests.get(call_url, params=call_params)
+
+    try:
+        extract_list = list(response.json()['query']['pages'].values())
+        return [
+            d['*'] for d in extract_list[0]['extlinks']
+        ]
+    except KeyError as e:
+        logger.error(f'Failed to parse Wikimedia JSON: {response.json()}')
+        raise e
+
+
 def is_in_references(references, doi):
     """ Search for a doi in a list of references.
 
     Args:
-        references (iter): References to loop through
-        doi (str): DOI of work to search for
+        references (iter): References to loop through.
```

Comment:
You could use `collections.abc.Iterable` here.
https://docs.python.org/3/library/collections.abc.html